src/plugins/dotplot/src/DotPlotWidget.cpp

```diff
@@ -1497,66 +1497,52 @@ void DotPlotWidget::selectNearestRepeat(const QPointF& p) {
     nearestSelecting = false;
 }
 
+float DotPlotWidget::calculateDistance(float x, float y, const DotPlotResults& r) const {
+    // find nearest point on line segment to desired point
+    // then calculate distance
+    double t = (double)(((x - r.x) + (y - r.y)) * r.len) / (2 * pow(r.len, 2));
```

Comment:
Could you please add a reference to the algorithm?
Like this: https://stackoverflow.com/questions/849211/shortest-distance-between-a-point-and-a-line-segment
or this
http://csharphelper.com/blog/2016/09/find-the-shortest-distance-between-a-point-and-a-line-segment-in-c/

++ If you develop a new algorithm by yourself please add API tests because this is a very error-prone algorithm
To make API tests simpler you can make this method static and pass all required parameters.

Reply:
Nearest point on line segment to desired point - https://www.cyberforum.ru/post16063107.html
Then we calculate distance between two points.

Reply:
Yes lets make this function an independent (static) at least. This is a very good candidate for a GeometryUtils-like class.